rpcd/playbooks/roles/elasticsearch/tasks/main.yml

```diff
@@ -13,9 +13,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+- include: upgrade_pre_flight.yml
+  when:
+    - logging_upgrade | bool
 - include: upgrade_elasticsearch_pre.yml
   when:
     - logging_upgrade | bool
+    - proceed_with_upgrade.ansible_facts.proceed_with_upgrade | bool
```

Comment:
Agreed, I'd be interested to see an example of where set_fact is not sufficient.